### PR TITLE
Metamorph: expand UIC1 tag parsing

### PR DESCRIPTION
**Pull request by melissalinkert**
Fixes https://trac.openmicroscopy.org/ome/ticket/12802.

To test, use the file from the ome-users thread linked to ticket 12802.  Check the original metadata table, and verify that the invalid entries noted in the thread are no longer present.

### CODE REVIEW

File: components/formats-gpl/src/loci/formats/in/MetamorphReader.java [hunk 1]
```diff
@@ -1030,6 +1030,9 @@ protected void initStandardMetadata() throws FormatException, IOException {
       if (uic4tagEntry != null) {
         mmPlanes = uic4tagEntry.getValueCount();
       }
+      if (mmPlanes == 0) {
+        mmPlanes = ifds.size();
```
Comment (sbesson) [inline components/formats-gpl/src/loci/formats/in/MetamorphReader.java]: Is it worth adding a comment explaining under what circumstances this block should be executed?

**Comment by sbesson** [comment]
Changes look good from a pure code review perspective. I will test its effect against data on Monday.

**Comment by sbesson** [comment]
Tested with the following command

```
diff <(~/bftools/showinf -nopix $1) <(~/bftools5.1.1/showinf -nopix $1)
```

where `bftools` is the unzipped command-line tools from the merge build and `bftools5.1.1` is the unzipped 5.1.1 command line tools.

For the file linked from the ome-users thread, the diff looks good

```
 $ ./diff_bftools b0BCDSL-T070_wD07_s4_z0_t1_cCy5_u001.tif 
7c7
< Initialization took 0.207s
---
> Initialization took 0.213s
70d69
< X position for position #1: 200599.04
72d70
< Y position for position #1: 4927.12
75a74
> AbsoluteZ: 2096309
83a83
> CameraChipOffset: 2096260
94c94
< Gamma: 1
---
> Gamma: 2096317
99a100
> PlaneProperty: 2097094
102a104
> StagePosition: 2096244
113d114
< UIC1 absoluteZ[0]: 11488.9
118,119d118
< cameraXChipOffset[0]: 0.0
< cameraYChipOffset[0]: 0.0
126,127d124
< stageX[0]: 200599.04
< stageY[0]: 4927.12
```

Similarly, the metadata of other MetaXPress files is now correctly parsed.

For Metamorph files however, this change seems to modify the metadata especially for the values of `camera[XY]ChipOffset` leading to extreme values, e.g.

```
$ ./diff_bftools /ome/data_repo/from_skyking/metamorph/jayne/2celltest10fpstl65power.stk
7c7
< Initialization took 0.519s
---
> Initialization took 0.506s
70d69
< X position for position #1: 2.808983409843019E-6
72d70
< Y position for position #1: 2.9092995760949875E-6
82a81
> CameraChipOffset: 9968035
90c89
< Gamma: 1
---
> Gamma: 9968075
96a96
> PlaneProperty: 9967980
99a100
> StagePosition: 9968019
169c170
< cameraXChipOffset[01]: 1.0
---
> cameraXChipOffset[01]: 0.0
172,186c173,187
< cameraXChipOffset[04]: 8.845643520566122E-4
< cameraXChipOffset[05]: 4.013036751189916E-7
< cameraXChipOffset[06]: 6.019545464122004E-7
< cameraXChipOffset[07]: 1.7976931348623157E308
< cameraXChipOffset[08]: 1.7976931348623157E308
< cameraXChipOffset[09]: 0.1875
< cameraXChipOffset[10]: 2.136263065537499E-4
< cameraXChipOffset[11]: 1.6052089028898776E-6
< cameraXChipOffset[12]: 2.5714285714285716
< cameraXChipOffset[13]: 1.7976931348623157E308
< cameraXChipOffset[14]: 2.2070583934480865E-6
< cameraXChipOffset[15]: 2.40770199792115E-6
< cameraXChipOffset[16]: 1.7976931348623157E308
< cameraXChipOffset[17]: 2.808983409843019E-6
< cameraXChipOffset[18]: 1.7976931348623157E308
---
> cameraXChipOffset[04]: 0.0
> cameraXChipOffset[05]: 0.0
> cameraXChipOffset[06]: 0.0
> cameraXChipOffset[07]: 0.0
> cameraXChipOffset[08]: 0.0
> cameraXChipOffset[09]: 0.0
> cameraXChipOffset[10]: 0.0
> cameraXChipOffset[11]: 0.0
> cameraXChipOffset[12]: 0.0
> cameraXChipOffset[13]: 0.0
> cameraXChipOffset[14]: 0.0
> cameraXChipOffset[15]: 0.0
> cameraXChipOffset[16]: 0.0
> cameraXChipOffset[17]: 0.0
> cameraXChipOffset[18]: 0.0
189,205c190,206
< cameraYChipOffset[02]: 1.0
< cameraYChipOffset[03]: 5.047955577990914E-4
< cameraYChipOffset[04]: 3.0
< cameraYChipOffset[05]: 5.016291912874634E-7
< cameraYChipOffset[06]: 7.022796700369068E-7
< cameraYChipOffset[07]: 0.03529411764705882
< cameraYChipOffset[08]: 0.0859375
< cameraYChipOffset[09]: 1.7976931348623157E308
< cameraYChipOffset[10]: 0.15
< cameraYChipOffset[11]: 1.7055330904520445E-6
< cameraYChipOffset[12]: 4.75
< cameraYChipOffset[13]: 2.106737557382264E-6
< cameraYChipOffset[14]: 2.307382933171268E-6
< cameraYChipOffset[15]: 1.7976931348623157E308
< cameraYChipOffset[16]: 1.7976931348623157E308
< cameraYChipOffset[17]: 2.9092995760949875E-6
< cameraYChipOffset[18]: 3.8121764728720004E-6
---
> cameraYChipOffset[02]: 0.0
> cameraYChipOffset[03]: 0.0
> cameraYChipOffset[04]: 0.0
> cameraYChipOffset[05]: 0.0
> cameraYChipOffset[06]: 0.0
> cameraYChipOffset[07]: 0.0
> cameraYChipOffset[08]: 0.0
> cameraYChipOffset[09]: 0.0
> cameraYChipOffset[10]: 0.0
> cameraYChipOffset[11]: 0.0
> cameraYChipOffset[12]: 0.0
> cameraYChipOffset[13]: 0.0
> cameraYChipOffset[14]: 0.0
> cameraYChipOffset[15]: 0.0
> cameraYChipOffset[16]: 0.0
> cameraYChipOffset[17]: 0.0
> cameraYChipOffset[18]: 0.0
251,288d251
< stageX[00]: 0.0
< stageX[01]: 0.0
< stageX[02]: 1.0
< stageX[03]: 0.0
< stageX[04]: 0.0
< stageX[05]: 8.845643520566122E-4
< stageX[06]: 4.013036751189916E-7
< stageX[07]: 6.019545464122004E-7
< stageX[08]: 1.7976931348623157E308
< stageX[09]: 1.7976931348623157E308
< stageX[10]: 0.1875
< stageX[11]: 2.136263065537499E-4
< stageX[12]: 1.6052089028898776E-6
< stageX[13]: 2.5714285714285716
< stageX[14]: 1.7976931348623157E308
< stageX[15]: 2.2070583934480865E-6
< stageX[16]: 2.40770199792115E-6
< stageX[17]: 1.7976931348623157E308
< stageX[18]: 2.808983409843019E-6
< stageY[00]: 0.0
< stageY[01]: 0.0
< stageY[02]: 0.0
< stageY[03]: 1.0
< stageY[04]: 5.047955577990914E-4
< stageY[05]: 3.0
< stageY[06]: 5.016291912874634E-7
< stageY[07]: 7.022796700369068E-7
< stageY[08]: 0.03529411764705882
< stageY[09]: 0.0859375
< stageY[10]: 1.7976931348623157E308
< stageY[11]: 0.15
< stageY[12]: 1.7055330904520445E-6
< stageY[13]: 4.75
< stageY[14]: 2.106737557382264E-6
< stageY[15]: 2.307382933171268E-6
< stageY[16]: 1.7976931348623157E308
< stageY[17]: 1.7976931348623157E308
< stageY[18]: 2.9092995760949875E-6
```

Other files that exhibit similar metadata modifications are `metamorph/jcb/201007118/si133_J3_5.nd` and `metamorph/josh/2101to400astk.stk`

**Comment by melissalinkert** [comment]
Commit pushed which I think will fix this.  The issue is that the MetaXpress files have camera offsets, positions, etc. in the UIC1 tag only, but the Metamorph files have camera offsets/positions in both the UIC1 and UIC4 tags.  The UIC4 tags are now preferred, so the Metamorph files should show UIC4 values and the MetaXpress files should show UIC1 values.

**Comment by sbesson** [comment]
Last commit works as expected both for Metamorph and MetaXpress files and removes the duplicate `UIC1AbsoluteZ` metadata in case `UIC4Absolute` exists. Looks ready to merge once this has been validated by the full-repository job.

**Pull request merged**